src/Microsoft.AspNetCore.Mvc.Core/Internal/MvcCoreLoggerExtensions.cs

```diff
@@ -639,7 +645,22 @@ public static IDisposable ActionScope(this ILogger logger, ActionDescriptor acti
 
         public static void ExecutingAction(this ILogger logger, ActionDescriptor action)
         {
-            _actionExecuting(logger, action.DisplayName, null);
+            var routeKeys = action.RouteValues.Keys.ToArray();
+            var routeValues = action.RouteValues.Values.ToArray();
+            string route = "\"";
+            for (var i = 0; i < routeValues.Count(); i++)
```

Comment:
I am not a huge fan of this. Any suggestions about how to format the route output?
Example output for this approach:
`Route "action:Index, controller:Home, page:" matched. Executing action MvcSandbox.Controllers.HomeController.Index (MvcSandbox)`

Reply:
I don't like how the null value for key "page" is an empty string

Reply:
I'd suggest a format like `{ action = "...", controller = ".." }` - this is what users type frequently in views, so it's familar